src/misc.c

```diff
@@ -256,10 +256,12 @@ void _libssh2_store_u32(unsigned char **buf, uint32_t value)
  */
 void _libssh2_store_str(unsigned char **buf, const char *str, size_t len)
 {
-    _libssh2_store_u32(buf, (uint32_t)len);
+    uint32_t len_stored = (uint32_t)len;
+
+    _libssh2_store_u32(buf, len_stored);
     if(len) {
```

Comment:
This should probably be `len_stored` instead of testing `len` to keep it consistent.